src/lib.rs

```diff
@@ -1,14 +1,12 @@
-pub fn add(left: u64, right: u64) -> u64 {
-    left + right
-}
+#![no_std]
 
-#[cfg(test)]
-mod tests {
-    use super::*;
+extern crate alloc;
+#[macro_use]
+extern crate log;
 
-    #[test]
-    fn it_works() {
-        let result = add(2, 2);
-        assert_eq!(result, 4);
-    }
-}
+pub use config::EmulatedDeviceConfig;
```

Comment:
Mind the sequence of `mod::xxx` and `pub use xxx`.

General you should put `mod::xxx` together, then  `pub use xxx` after a blank line.